File: backend/src/main/java/com/backend/domain/user/repository/UserRepository.java
package com.backend.domain.user.repository;

import com.backend.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User,Long> {
    Optional<User> findByEmail(String email);

    @Query("SELECT u FROM User u WHERE u.email = :email AND u.deleted = true")
    Optional<User> findByEmailIncludeDeleted(@Param("email") String email);

    boolean existsByEmail(String email);

    User findNameById(Long userId);
}

File: backend/src/main/java/com/backend/domain/user/repository/UserRepository.kt
package com.backend.domain.user.repository

import com.backend.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    //Optional대신 nullable 타입반환
    fun findByEmail(email: String): User?

    /*

    @Query("SELECT u FROM User u WHERE u.email = :email AND u.isDeleted = true")
    Optional<User> findByEmailIncludeDeleted(@Param("email") String email);
*/
    fun existsByEmail(email: String): Boolean

    fun findNameById(userId: Long?): User?
}
